dojo/problem/redis.py

```diff
@@ -0,0 +1,142 @@
+import base64
+import json
+import logging
+from dataclasses import dataclass, field
+from functools import lru_cache
+
+import redis
+
+import dojo.problem.helper as problems_help
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+SEVERITY_ORDER = {
+    "Critical": 5,
+    "High": 4,
+    "Medium": 3,
+    "Low": 2,
+    "Info": 1,
+}
+
+
+@dataclass
+class Problem:
+    problem_id: str
+    name: str = ""
+    severity: str = "Info"
+    script_ids: set = field(default_factory=set)
+    finding_ids: set = field(default_factory=set)
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "problem_id": self.problem_id,
+            "severity": self.severity,
+            "script_ids": list(self.script_ids),
+            "finding_ids": list(self.finding_ids),
+        }
+
+    @staticmethod
+    def from_dict(data):
+        return Problem(
+            name=data["name"],
+            problem_id=data["problem_id"],
+            severity=data["severity"],
+            script_ids=set(data.get("script_ids", [])),
+            finding_ids=set(data.get("finding_ids", [])),
+        )
+
+    @classmethod
+    def load_from_id(cls, problem_id, redis_client):
+        problem_data = redis_client.hget("problems", problem_id)
+        if problem_data:
+            return cls.from_dict(json.loads(problem_data))
+        return None
+
+    def add_finding(self, finding):
```

Comment:
This function only seems to be called from `ProblemHelper.add_finding()`. What is the reasoning behind having two classes? Why is having two classes preferrable to having all the code in a single place? In other words, to we ever need `Problem.add_finding()` without `ProblemHelper.add_finding()`?